ai/Algorithm/evaluation_module.py

```diff
@@ -25,14 +26,80 @@ def player_with_ball(min_dist_from_ball=1.2*ROBOT_RADIUS, our_team=None):
+def player_pointing_toward_point(player: Player, point: Position, angle_tolerance=90 * np.pi / 180):
+    return abs((player.pose.orientation - (point - player.position).angle)) < angle_tolerance / 2
+
+
+def object_pointing_toward_point(object_position, object_orientation, point, angle_tolerance=90 * np.pi / 180):
+    return abs((object_orientation - (point - object_position).angle)) < angle_tolerance / 2
 
 
 def player_pointing_toward_segment(player: Player, segment: Line):
-    angle_biscetion = angle_between_three_points(segment.p1, player.position, segment.p2) / 2
-    angle_reference = angle_biscetion + (segment.p2 - player.position).angle
-    return abs(player.pose.orientation - angle_reference) < angle_biscetion
+    angle_bisection = angle_between_three_points(segment.p1, player.position, segment.p2) / 2
+    angle_reference = angle_bisection + (segment.p2 - player.position).angle
+    return abs(player.pose.orientation - angle_reference) < angle_bisection
+
+
+def player_covered_from_goal(player: Player):
+    shooting_angle = angle_between_three_points(GameState().field.their_goal_line.p1,
+                                                player.position, GameState().field.their_goal_line.p2)
+    vec_player_to_goal = GameState().field.their_goal - player.position
+
+    our_team = [other_player for other_player in GameState().our_team.available_players.values() if other_player is not player]
+    enemy_team = [other_player for other_player in GameState().enemy_team.available_players.values()]
+    pertinent_collisions = []
+    for other_player in our_team + enemy_team:
+        if object_pointing_toward_point(player.position,
+                                        vec_player_to_goal.angle,
+                                        other_player.position,
+                                        wrap_to_pi(shooting_angle + 5 * np.pi / 180)):
+            pertinent_collisions.append(Obstacle(other_player.position.array, avoid_distance=90))
+
+    if not any(pertinent_collisions):
+        return GameState().field.their_goal
+    pertinent_collisions_positions = np.array([obs.position for obs in pertinent_collisions])
+    pertinent_collisions_avoid_radius = np.array([obs.avoid_distance for obs in pertinent_collisions])
+    results = []
+    for i in range(1, 20):  # discretisation de la ligne de but
+        goal_point = GameState().field.their_goal_line.p1 + GameState().field.their_goal_line.direction * \
+                     (GameState().field.their_goal_line.length * i / 20)
+        is_colliding = is_path_colliding(pertinent_collisions, pertinent_collisions_positions,
+                                         pertinent_collisions_avoid_radius, player.position.array, goal_point.array)
+        results.append([is_colliding, goal_point])
+    count = 0
+    max_len_seg = 0 #longueur du segment
+    indexend = 0
+    for i, result in enumerate(results):
```

Comment:
Tu peux faire `for i, is_colliding, _ in enumerate(results):`, si tu utilise pas le goal_point

Reply:
met 20 comme variable please

Reply:
Aussi, creer une fonction `find_max_length_segment` ou un truc du genre pour clarifier